api/v1alpha1/amaltheasession_types.go

```diff
@@ -0,0 +1,265 @@
+	TLSSecretName string `json:"tlsSecretName,omitempty"`
+}
+
+type Storage struct {
+	ClassName string            `json:"storageClassName,omitempty"`
+	Size      resource.Quantity `json:"storageSize,omitempty"`
+	// The absolute mount path for the session volume
+	// +kubebuilder:default:=/workspace
+	MountPath string `json:"mountPath,omitempty"`
+}
+
+type CodeRepository struct {
+	// +kubebuilder:example:=repositories/project1
+	// +kubebuilder:default:="."
+	// Path relative to the session working directory where the repository should be cloned into.
+	ClonePath string `json:"clonePath,omitempty"`
+	// +kubebuilder:example:="https://github.com/SwissDataScienceCenter/renku"
+	// The HTTP url to the git repository
+	Remote string `json:"remote"`
+	// +kubebuilder:example:=main
+	// The tag, branch or commit SHA to checkout, if ommitted then will be the tip of the default branch of the repo
+	Revision string `json:"revision,omitempty"`
+	// The Kubernetes secret that contains the git configuration to be used during cloning.
+	// This can be used to inject credentials in addition to any other repo-specific Git configuration.
+	// NOTE: you have to specify the whole git config in a single key in the secret.
+	CloningGitConfigSecretRef SecretRef `json:"cloningGitConfigSecretRef,omitempty"`
+	// The Kubernetes secret that contains the git configuration to be used when the session is running.
+	// This can be used to inject credentials in addition to any other repo-specific Git configuration.
+	// NOTE: you have to specify the whole git config in a single key in the secret.
+	GitConfigSecretRef SecretRef `json:"gitConfigSecretRef,omitempty"`
+}
+
+// +kubebuilder:validation:Enum={rclone}
+type StorageType string
+
+const Rclone StorageType = "rclone"
+
+type DataSource struct {
+	// +kubebuilder:default:=rclone
+	// The data source type
+	Type StorageType `json:"type,omitempty"`
+	// +kubebuilder:example:=data/storages
+	// +kubebuilder:default:="."
+	// Path relative to the session working directory where the data should be mounted
+	MountPath string `json:"mountPath,omitempty"`
+	// The secret containing the configuration or credentials needed for access to the data.
+	// The format of the configuration that is expected depends on the storage type.
+	// NOTE: define all values in a single key of the Kubernetes secret.
+	// rclone: any valid rclone configuration for a single remote, see the output of `rclone config providers` for validation and format.
+	SecretRef string `json:"secretRef,omitempty"`
+}
+
+type Culling struct {
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	// The maximum allowed age for a session, regardless of whether it
+	// is active or not. When the threshold is reached the session is hibernated.
+	// A value of zero indicates that Amalthea will not automatically hibernate
+	// the session based on its age.
+	MaxAgeSeconds int `json:"maxAgeSeconds,omitempty"`
```

Comment:
I am wondering whether we should use a better defined version of `int` for these type of duration field such as `int32` or `int64` since `int` can be either 32 or 64 bit depending on the architecture.

Reply:
K8s has a type for duration - so I used that. It will allow us to specify things like `2h5m4s`.

Reply:
AFAIK even the base time package has a `Duration` type and a function to parse a string such the one above: https://pkg.go.dev/time#ParseDuration

Reply:
@aledegano yes and they have a type for the already in the k8s api that I can use directly and easily. So I am using that. Thanks!